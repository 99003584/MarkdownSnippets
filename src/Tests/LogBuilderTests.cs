using System.Collections.Generic;
using ApprovalTests;
using MarkdownSnippets;
using Xunit;
using Xunit.Abstractions;

public class LogBuilderTests :
    XunitLoggingBase
{
    [Fact]
    public void BuildConfigLogMessage()
    {
        var config = new ConfigResult
        {
            WriteHeader = true,
            Header = @"line1
line2",
            Exclude = new List<string> {"Dir1", "Dir2"},
            ReadOnly = true,
            LinkFormat = LinkFormat.Tfs,
            UrlsAsSnippets = new List<string> {"Url1", "Url2"},
            TocLevel = 5
        };
        var message = LogBuilder.BuildConfigLogMessage("theRoot", config, "theConfigFilePath");
        Approvals.Verify(message);
    }

    [Fact]
    public void BuildConfigLogMessageMinimal()
    {
        var config = new ConfigResult();
        var message = LogBuilder.BuildConfigLogMessage("theRoot", config, "theConfigFilePath");
        Approvals.Verify(message);
    }

    public LogBuilderTests(ITestOutputHelper output) :
        base(output)
    {
    }
}