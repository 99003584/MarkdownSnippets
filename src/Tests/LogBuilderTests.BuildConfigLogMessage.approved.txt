Config:
    RootDir: theRoot
    ReadOnly: True
    WriteHeader: True
    Header:
        line1
        line2
    LinkFormat: Tfs
    TocLevel: 5
    FileConfigPath: theConfigFilePath (exists:False)
    Exclude:
        Dir1
        Dir2
    UrlsAsSnippets:
        Url1
        Url2